src/StoreMeta.h

```diff
@@ -80,67 +64,96 @@ enum {
         uint16_t flags;
      \endcode
      */
-    STORE_META_STD,
-
-    /**
-     * Reserved for future hit-metering (RFC 2227) stuff
-     */
-    STORE_META_HITMETERING,
-
-    // TODO: document this TLV type code
-    STORE_META_VALID,
+    STORE_META_STD = 5,
 
     /**
      * Stores Vary request headers
      */
-    STORE_META_VARY_HEADERS,
+    STORE_META_VARY_HEADERS = 8,
 
     /**
      * Updated version of STORE_META_STD, with support for  >2GB objects.
      * As STORE_META_STD except that the swap_file_sz is a 64-bit integer instead of 32-bit.
      */
-    STORE_META_STD_LFS,
+    STORE_META_STD_LFS = 9,
 
     // TODO: document this TLV type code
-    STORE_META_OBJSIZE,
+    STORE_META_OBJSIZE = 10
 
-    STORE_META_STOREURL,    /**< the Store-ID url, if different to the normal URL */
-    STORE_META_VARY_ID,     /**< Unique ID linking variants */
-    STORE_META_END
+    // When adding values, update Store::SwapMetaTypeMax.
+    // When removing values, check Store::IgnoredSwapMetaType() and friends.
 };
 
-/// \ingroup SwapStoreAPI
-class StoreMeta
+namespace Store {
+
+/// swap meta type ID written to or loaded from Store
+using RawSwapMetaType = char;
+
+/// maximum value of a recognized swap meta type
+const int SwapMetaTypeMax = 12; // Use "inline constexpr ..." with C++17.
+
+/// Store entries with longer swap metadata field values are not swapped out and
+/// are considered invalid when validating being-loaded metadata. This arbitrary
+/// limit protects code that adds individual swap metadata field sizes from
+/// overflowing and also prevents allocation of huge buffers when loading
+/// variable-length fields. Reevaluate this limit when increasing MAX_URL.
+const size_t SwapMetaFieldValueLengthMax = 64*1024;
+
+// TODO: Move to src/store/SwapMetaReading or similar, along with StoreMetaUnpacker.
```

Comment:
Per coding guidelines this should be in store/SwapMetaView.h and the unpacker in store/StoreMetaUnpacker.h

Reply:
Addressed in 84a0f8d: SwapMetaView got its own source files, and StoreMetaUnpacker was converted into an internal helper class that the rest of the code does not need to be bothered with or exposed to.